src/main/java/com/cglx/Calculator.java

```diff
@@ -31,9 +37,25 @@ public int square(int a)
         return a * a;
     }
 
-    private void readFile() {
-        FileInputStream fis = new FileInputStream(new File("TestFile.txt"));
-        
+    private List<String> readFile(String filename)
+    {
+      List<String> records = new ArrayList<String>();
+      try
+      {
+        BufferedReader reader = new BufferedReader(new FileReader(filename));
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Close this "FileReader" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-01.ahprod.adhaven.com:9990/coding_rules#rule_key=squid%3AS2095)